requirements.txt

```diff
@@ -5,6 +5,7 @@ torch>=1.3
 future>=0.17.1  # required for builtins in setup.py
 # pyyaml>=3.13
 PyYAML>=5.1  # OmegaConf requirement >=5.1
+jsonargparse[signatures]>=3.0.1  # LightningCLI requirement
```

Comment:
Is this required for the Lightning installation or can it be moved to extras?

Reply:
Right now the argument parser and the cli class are completely independent from everything else. So the jsonargparse requirement could be moved to an extra without any issue. However, I don't know if you might want to use the new argument parser to replace some of the old parser functionality. Depending on that this requirement could be needed here. It all depends on what the core contributors decide. Based on what you want I could advice on different options.

Reply:
Any further comment on this?

Reply:
> I don't know if you might want to use the new argument parser to replace some of the old parser functionality

If this PR lands, we will want to replace the old one. The PRs would be:

Addition (this one) -> Deprecation of current implementation -> Deletion of current implementation

So I think this PR should include the install in extras and the deletion PR should move it to the base requirements.

Reply:
I moved the requirement to extras. But now some doctests are failing because jsonargparse is not found. I have no idea how to fix this.

Reply:
@Borda can you help?

Reply:
yeah, I ll plan to spend Monday to get this over the finish line :]